### PR TITLE
feat(ci): enable nightly e2e tests

### LINKED ISSUE #262: Packet E2E test

### PR DESCRIPTION
**Pull request by rsmitty**
Will close #262

### CODE REVIEW

File: hack/test/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+set -eou pipefail
+
+export TMP="$(mktemp -d)"
+export OSCTL="${PWD}/build/osctl-linux-amd64"
+export TALOSCONFIG="${TMP}/talosconfig"
+export KUBECONFIG="${TMP}/kubeconfig"
+
+##ClusterAPI Provider Talos (CAPT)
```
Comment (andrewrynhard) [inline hack/test/e2e.sh]: Total nit: 1 `#` plus  a space between the text:
```
# ClusterAPI Provider Talos (CAPT)
```

File: hack/test/e2e.sh [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	         exit 1
+	       fi
+           sleep 5
+         done"
+
+# ## Wait for 2 nodes to check in
+e2e_run "timeout=\$((\$(date +%s) + ${TIMEOUT}))
+         until KUBECONFIG=${KUBECONFIG}-capi kubectl get nodes -o json | jq '.items | length' | grep ${NUM_NODES} >/dev/null
+         do 
+		   if  [[ \$(date +%s) -gt \$timeout ]]
```
Comment (andrewrynhard) [inline hack/test/e2e.sh]: nit: Align `if` with the `then`.

File: hack/test/e2e.sh [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+export KUBECONFIG="${TMP}/kubeconfig"
+
+##ClusterAPI Provider Talos (CAPT)
+CAPT_VERSION="0.1.0-alpha.1"
+PROVIDER_COMPONENTS="https://github.com/talos-systems/cluster-api-provider-talos/releases/download/v${CAPT_VERSION}/provider-components.yaml"
+KUSTOMIZE_VERSION="1.0.11"
+KUSTOMIZE_URL="https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZE_VERSION}/kustomize_${KUSTOMIZE_VERSION}_linux_amd64"
+
+##Total number of nodes we'll be waiting to come up
+NUM_NODES=2
```
Comment (andrewrynhard) [inline hack/test/e2e.sh]: Not for this PR, but we should follow up with a PR to make this test HA.

File: hack/test/e2e.sh (hunk 2 above)
Comment (rsmitty) [reply hack/test/e2e.sh]: Should be good now. Was just some tabs vs. space nonsense in vscode

File: hack/test/e2e.sh (hunk 1 above)
Comment (rsmitty) [reply hack/test/e2e.sh]: Fixed

**Comment by smira** [comment]
Does this bootstrap docker cluster with cluster API to build packet cluster?

**Comment by rsmitty** [comment]
> Does this bootstrap docker cluster with cluster API to build packet cluster?

Yep, exactly. It should give us a good chance to do a full end to end everyday, including a release version of our CAPI provider.

**Pull request merged**